bluemira/base/reactor_config.py

```diff
@@ -231,30 +227,31 @@ def _check_args_are_strings(self, args: Iterable[str]):
             if not isinstance(a, str):
                 raise ReactorConfigError("args must be strings")
 
-    def _get_nested_filepaths(
-        self,
-        current_layer: str,
-        current_arg_key: str,
-        arg_keys: Tuple[str],
-        next_idx: int,
-    ) -> Dict:
-        if not Path(current_layer).resolve().is_file():
-            if isinstance(self._config_path, str):
-                current_layer = str(Path(self._config_path).parent / current_layer)
-            if not Path(current_layer).resolve().is_file():
-                raise FileNotFoundError(f"Cannot find file {current_layer}")
-
-        new_layer = self._read_json_file(current_layer)
-
-        # Dont need to load files repeatedly so put it back in config_data
-        if next_idx == 1:
-            self.config_data[current_arg_key] = new_layer
-        else:
-            tmp_dict = self.config_data
-            for key in arg_keys[: next_idx - 1]:
-                tmp_dict = tmp_dict[key]
-            tmp_dict[current_arg_key] = new_layer
-        return new_layer
+    def _extract_file_data_if_needed(self, value: Any) -> Union[Any, dict]:
+        if not isinstance(value, str):
+            return value
+        if not value.startswith(_FILEPATH_PREFIX):
+            return value
+
+        # remove _FILEPATH_PREFIX
+        f_path = value[len(_FILEPATH_PREFIX) :]
+
+        # replace . with a path relative to the config file
+        if f_path.startswith("."):
+            if not isinstance(self._config_path, str):
+                raise ReactorConfigError(
+                    "Can only use relative paths with a path config"
+                )
+            # handles removing the first . and the /
+            f_path = Path(self._config_path).parent / f_path
+
+        # check if file exists
+        if not Path(f_path).is_file():
+            raise FileNotFoundError(f"Cannot find file {f_path}")
```

Comment:
why not just use `Path(path).resolve()`  and then `.is_file()` check if config_path is not set and raise if `is_file()` is false?  This allows for a little bit more flexibility and offloads the relative path checking to `Path`
```suggestion
        f_path = Path(f_path).resolve()
        if not f_path.is_file():
            if not isinstance(self._config_path, str):
                raise ReactorConfigError(
                    "Can only use relative paths with a path config"
                )
            # handles removing the first . and the /
            f_path = Path(self._config_path).parent / f_path
            
            if not f_path.is_file():
                raise FileNotFoundError(f"Cannot find file {f_path}")
```

Reply:
You wouldn't want to check if f_path is a file until you've constructed the full (intended) path.
In your suggestion, after the call to .resolve(), that could point to a file in the bluemira dir. which is what's intended by the relative path.

Reply:
I agree though that the way I wrote it is a bit weird with f_path being either a `str` or `Path` when it's used again in the `not Path(f_path).is_file()`
So will change